network/p2p/test_utils.go

```diff
@@ -69,44 +66,59 @@ func (ln *LocalNet) WithBootnode(ctx context.Context, logger *zap.Logger) error
 }
 
 // CreateAndStartLocalNet creates a new local network and starts it
-func CreateAndStartLocalNet(pctx context.Context, loggerFactory LoggerFactory, forkVersion forksprotocol.ForkVersion, n, minConnected int, useDiscv5 bool) (*LocalNet, error) {
-	ln, err := NewLocalNet(pctx, loggerFactory, n, forkVersion, useDiscv5)
-	if err != nil {
-		return nil, err
-	}
-	var wg sync.WaitGroup
-	var peersErr error
-	for i, node := range ln.Nodes {
-		wg.Add(1)
-		go func(node network.P2PNetwork, i int) {
-			logger := loggerFactory(fmt.Sprintf("node-%d", i))
-			if err := node.Start(); err != nil {
-				logger.Warn("could not start node", zap.Error(err))
-				wg.Done()
-				return
-			}
-			go func(node network.P2PNetwork, logger *zap.Logger) {
-				defer wg.Done()
-				ctx, cancel := context.WithTimeout(pctx, 15*time.Second)
+// if any errors occurs during starting local network CreateAndStartLocalNet trying
+// to create and start local net one more time until pCtx is not Done()
+func CreateAndStartLocalNet(pCtx context.Context, logger *zap.Logger, forkVersion forksprotocol.ForkVersion, nodesQuantity, minConnected int, useDiscv5 bool) (*LocalNet, error) {
+	attempt := func(pCtx context.Context, forkVersion forksprotocol.ForkVersion, nodesQuantity, minConnected int, useDiscv5 bool) (*LocalNet, error) {
+		ln, err := NewLocalNet(pCtx, logger, nodesQuantity, forkVersion, useDiscv5)
+		if err != nil {
+			return nil, err
+		}
+
+		eg, ctx := errgroup.WithContext(pCtx)
+		for i, node := range ln.Nodes {
+			i, node := i, node //hack to avoid closures. price of using error groups
+
+			eg.Go(func() error {
+				if err := node.Start(); err != nil {
+					return fmt.Errorf("could not start node %d: %w", i, err)
+				}
+				ctx, cancel := context.WithTimeout(ctx, 1000*time.Second)
```

Comment:
Is `1000*time.Second` intentional? It's 16.7 minutes

Reply:
it's a mistake, fixed